### PR TITLE
Add toy problem README with problem details

### PR DESCRIPTION
**Pull request by AlbertLee125**
I added the toy problem from Linan's paper to the gdplib.

### CODE REVIEW

**Comment by ZedongPeng** [comment]
The black format check failed. @AlbertLee125

**Comment by AlbertLee125** [comment]
The toy problem is ready to go. I changed the title to david_linan.py and changed the directory to david_linan. The spelling is failing because of the HDA_gdpopt file. Besides the hda file error, the code is ready to go.

Second thoughts: Gomez_linan is also a good name for the toy problem. They are the ones to create the given model. If you both think the title is better written as 'gomez_linan.py', please let me know.

File: gdplib/david_linan/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## david_linan.py
```
Comment (bernalde) [inline gdplib/david_linan/README.md]: Not a good name. Use ex1_linan_2023.py

File: gdplib/david_linan/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+The toy problem is a simple optimization problem that involves two Boolean variables, two continuous variables, and a non-linear objective function.
```
Comment (bernalde) [inline gdplib/david_linan/README.md]: This is no longer the "toy problem"

File: gdplib/david_linan/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+The problem is formulated as a Generalized Disjunctive Programming (GDP) model.
+The Boolean variables are associated with disjuncts that define the feasible regions of the continuous variables.
+The problem also includes logical constraints that ensure that only one Boolean variable is true at a time.
+The objective function is -0.9995999999999999 when the continuous variables are alpha = 0 (Y1[2]=True) and beta=-0.7 (Y2[3]=True).
+
+The objective function of the model is originated from Problem No. 6 of Gomez's paper and Linan introduced logic proposition, logic disjunctions and the following eqautinons as the constraints.
```
Comment (bernalde) [inline gdplib/david_linan/README.md]: There are typos and misspellings in this paragraph.

File: gdplib/david_linan/david_linan.py
Comment (bernalde) [inline gdplib/david_linan/david_linan.py:1]: See comments from other file

**Review by bernalde — CHANGES_REQUESTED** [review]
Change all parts where it is referred to 'Toy problem'.
Does this run with the ldsda implementation?

**Comment by AlbertLee125** [comment]
> Change all parts where it is referred to 'Toy problem'. Does this run with the ldsda implementation?

```
m = build_model()
    pyo.SolverFactory('gdpopt.ldsda').solve(
        m,
        minlp_solver="gams",
        minlp_solver_args=dict(solver="baron", add_options=["option optcr=0.001;"]),
        tee=True,
        starting_point=[5, 1],
        logical_constraint_list=[m.oneY1, m.oneY2],
        direction_norm='Linf',
        time_limit=900
    )
    print("Solution: alpha=", pyo.value(m.alpha), " beta=", pyo.value(m.beta))
    print("Objective function value: ", pyo.value(m.obj))
   ```
   If I set the starting point at [5,1], the code could not cross the infeasible wall of the problem. However, by setting the starting point at [1,1], the code could find the global minimum. I will mention about the infeasible wall in the documentation.

File: gdplib/ ex1_linan_2023/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+## ex1_linan.py
```
Comment (bernalde) [inline gdplib/ ex1_linan_2023/README.md]: This is not the name of the file

File: gdplib/ ex1_linan_2023/README.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+The `ex1_linan.py` file is a simple optimization problem that involves two Boolean variables, two continuous variables, and a nonlinear objective function. The problem is formulated as a Generalized Disjunctive Programming (GDP) model.
```
Comment (bernalde) [inline gdplib/ ex1_linan_2023/README.md]: This is not the name of the file

**Review by bernalde — CHANGES_REQUESTED** [review]

**Comment by AlbertLee125** [comment]
I changed the title and name as Example 1 Problem of Liñán (2023).

**Review by bernalde — APPROVED** [review]

**Pull request merged**